portfolio/src/main/webapp/SlideShow.js

```diff
@@ -0,0 +1,78 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/** The SlideShow class maintains the slideshows on the website. */
+class SlideShow {
+  /**
+   * Operates on an instance of SlideShow.
+   * @param {?HTMLCollection} collectionOfSlides Holds a collection of slides
+   *     from the webpage, defined by a class tag.
+   */
+  constructor(collectionOfSlides) {
+    /** @private {?HTMLCollection} */
+    this.slides_ = collectionOfSlides;
+
+    /** @private {number} */
+    this.slideIndex_ = 0;
+
+    /** @private @const {number} */
+    this.SLIDE_LENGTH_ = collectionOfSlides.length;
```

Comment:
No need for this variable. You can always get the length by calling collectionOfSlides.length